### PR TITLE
8263538: SharedArchiveConsistency.java should test -Xshare:auto as well

### PR DESCRIPTION
**Pull request by yminqi**
Hi, please review

  When this test run with -Xshare:on, it will exit when the shared archive is corrupted or mismatch. Make it run with -Xshare:auto will expand code coverage and to keep app run without using the CDS.

  Tests: Mach5 tier1,tier4

Thanks
Yumin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263538](https://bugs.openjdk.java.net/browse/JDK-8263538): SharedArchiveConsistency.java should test -Xshare:auto as well

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4694/head:pull/4694` \
`$ git checkout pull/4694`

Update a local copy of the PR: \
`$ git checkout pull/4694` \
`$ git pull https://git.openjdk.java.net/jdk pull/4694/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4694`

View PR using the GUI difftool: \
`$ git pr show -t 4694`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4694.diff">https://git.openjdk.java.net/jdk/pull/4694.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java [hunk 1]
```diff
@@ -397,36 +411,50 @@ public static void main(String... args) throws Exception {
...
-        System.out.println("\n2b. Corrupt _jvm_ident run with -Xshare:auto\n");
-        output = TestCommon.execAuto(execArgs);
+        // use the same archive as above, but run with -Xshare:auto or -Xshare:on depends on shareAuto (reverse)
+        System.out.println("\n2b. Corrupt _jvm_ident run with " + (shareAuto ? "-Xshare:on\n" : "-Xshare:auto"));
+        output = shareAuto ? TestCommon.execCommon(execArgs) : TestCommon.execAuto(execArgs);
         output.shouldContain("The shared archive file was created by a different version or build of HotSpot");
-        output.shouldContain("Hello World");
+        if (!shareAuto) {
+          output.shouldContain("Hello World");
+        }
```
Comment (iklam) [inline test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java:437]: Are lines 431-437 still necessary, now that both -Xshare:on and -Xshare:off are already executed for this test scenario.

Comment (yminqi) [reply test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java:437]: I was thinking of that either, the comment said using same archive. The other run with new created archive ---- though they have same contents. I could delete this part.

**Review by iklam — APPROVED** [review]
LGTM.

**Review by calvinccheung — APPROVED** [review]
Looks good.

**Comment by yminqi** [comment]
@iklam @calvinccheung Thanks for review!
/integrate

**Pull request closed without merging**